Client/Frontend/Shields/ShieldsViewController.swift

```diff
@@ -0,0 +1,164 @@
+        shieldControlMapping.forEach { shield, view, option in
+            if let option = option {
+                // Sets the default setting
+                view.toggleSwitch.isOn = option.value
+            } else {
+                // The only "shield" that doesn't actually have a global preference is the option to disable all shields
+                // Therefore its the only shield we have to give a default value of true
+                view.toggleSwitch.isOn = shield == .AllOff
+            }
+            if let host = url.normalizedHost, let shieldState = BraveShieldState.getStateForDomain(host) {
+                // Sets the site-specific setting
+                if var shieldOverrideEnabled = shieldState.isShieldOverrideEnabled(shield) {
+                    if shield == .AllOff {
+                        // Reversed...
+                        shieldOverrideEnabled = !shieldOverrideEnabled
+                        // TODO: Switch ^ to `isEnabled.toggle()` when we support Swift 4.2+
+                    }
+                    view.toggleSwitch.isOn = shieldOverrideEnabled
+                }
+            }
+        }
+        updateGlobalShieldState(shieldsView.shieldOverrideControl.toggleSwitch.isOn)
+    }
+    
+    private func updateShieldBlockStats() {
+        guard let stats = shieldBlockStats else {
+            // All 0
+            shieldsView.shieldsContainerStackView.adsTrackersStatView.valueLabel.text = "0"
+            shieldsView.shieldsContainerStackView.httpsUpgradesStatView.valueLabel.text = "0"
+            shieldsView.shieldsContainerStackView.scriptsBlockedStatView.valueLabel.text = "0"
+            shieldsView.shieldsContainerStackView.fingerprintingStatView.valueLabel.text = "0"
+            return
+        }
+        shieldsView.shieldsContainerStackView.adsTrackersStatView.valueLabel.text = String(stats.adsAndTrackers)
+        shieldsView.shieldsContainerStackView.httpsUpgradesStatView.valueLabel.text = String(stats.httpsUpgrades)
+        shieldsView.shieldsContainerStackView.scriptsBlockedStatView.valueLabel.text = String(stats.blockedScripts)
+        shieldsView.shieldsContainerStackView.fingerprintingStatView.valueLabel.text = String(stats.fingerprintingProtection)
+    }
+    
+    private func updateBraveShieldState(shield: BraveShieldState.Shield, on: Bool) {
+        guard let domain = url.normalizedHost else { return }
+        if shield == .AllOff {
+            // Technically we set "all off" when the switch is OFF, unlike all the others
+            BraveShieldState.set(forDomain: domain, state: (.AllOff, !on))
+        } else {
+            BraveShieldState.set(forDomain: domain, state: (shield, on))
+        }
+    }
+    
+    private func updateGlobalShieldState(_ on: Bool, animated: Bool = false) {
+        let updateBlock = {
+            self.shieldsView.shieldsContainerStackView.isHidden = !on
+            self.shieldsView.shieldsContainerStackView.alpha = on ? 1.0 : 0.0
```

Comment:
Any reason why we set both `isHidden` and `alpha` values here?

Reply:
Just animation purposes, looks nicer when its alpha fading at the same time rather than just being clipped while it does the `isHidden` animation.

Reply:
Any reason we need to use `isHidden` at all?